src/cis/fam_cis_direct.cpp

```diff
@@ -426,17 +509,17 @@ Fam_Region_Item_Info Fam_CIS_Direct::allocate(string name, size_t nbytes,
     dataitem.gid = gid;
     dataitem.uid = uid;
     dataitem.size = nbytes;
+    dataitem.memoryServerId = id;
     if (name == "") {
         metadataService->metadata_insert_dataitem(dataitemId, regionId,
                                                   &dataitem);
     } else {
         metadataService->metadata_insert_dataitem(dataitemId, regionId,
                                                   &dataitem, name);
     }
-    if (check_dataitem_permission(dataitem, 1, memoryServerId, uid, gid)) {
+    if (check_dataitem_permission(dataitem, 1, 0, uid, gid)) {
```

Comment:
better to use metadataserverId here also. Search for all occurance of check_dataitem_permission.